File: C/14_Backtracking/BOJ_03_15651_N과M(3).c
#include <stdio.h>
int arr[8], n, m;
void dfs(int dpth) {
	if (dpth == m) {
		for (int i = 0; i < m; i++) {
			printf("%d ", arr[i]);
		}
		printf("\n");
		return;
	}
	for (int i = 0; i < n; i++) {
		arr[dpth] = i + 1;
		dfs(dpth + 1);
	}
}
main() {
	scanf("%d %d", &n, &m);
	dfs(0);
}

File: JAVA/14_Backtracking/BOJ_03_15651_N과M(3).java
import java.io.IOException;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

    static int n, m;
    static int[] arr;
    static StringBuilder sb;

    public static void dfs(int dpth) {
        if (dpth == m) {
            for (int i : arr) {
                sb.append(i).append(" ");
            }
            sb.append("\n");
            return;
        }
        for (int i = 0; i < n; i++) {
            arr[dpth] = i + 1;
            dfs(dpth + 1);
        }
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine(), " ");
        n = Integer.parseInt(st.nextToken());
        m = Integer.parseInt(st.nextToken());
        arr = new int[m];
        sb = new StringBuilder();
        dfs(0);
        System.out.print(sb);
    }
}